nixos/modules/services/misc/jackett.nix

```diff
@@ -60,6 +60,56 @@ in
         Group = cfg.group;
         ExecStart = "${cfg.package}/bin/Jackett --NoUpdates --DataFolder '${cfg.dataDir}'";
         Restart = "on-failure";
+        
+        # Sandboxing
+        CapabilityBoundingSet = [
+          "CAP_NET_BIND_SERVICE"
+        ];
+        ExecPaths = [
+          "${builtins.storeDir}"
```

Comment:
This resolves to `/nix/store`, making that the only path where Jackett can execute binaries.

Reply:
Does Jackett even run binaries? I think the only thing would be its updater?

Reply:
Sorry, I haven't noticed this one.
If I remember correctly it does not start if I leave `ExecPath` empty. I assume that's because the `jackett` binary is wrapped in a Bash script (`Jackett`) which is run by the Systemd service.
Though I agree about the updater part.

Unfortunately I can't try it out at the moment as Nix tries to rebuild Jackett from source and I've got 2 tests failing:
```
Test Run Failed.
Total tests: 125
     Passed: 123
     Failed: 2
 Total time: 3.1566 Seconds
error: builder for '/nix/store/9ldqqgn50faafn5l27j871yf2wc4qd3q-jackett-0.20.2264.drv' failed with exit code 1

```

Reply:
I'm on the same. The problem is they wrote flaky tests depending on the actual year. 

For example:
```
Failed ParseDateTimeGoLangTest [4 ms]
  Error Message:
     Expected: 2021-09-14 07:00:00
  But was:  2022-09-14 07:00:00
```
https://github.com/Jackett/Jackett/blob/v0.20.2264/src/Jackett.Test/Common/Utils/DateTimeUtilTests.cs#L154

If you look at the [same test on `master`](https://github.com/Jackett/Jackett/blob/master/src/Jackett.Test/Common/Utils/DateTimeUtilTests.cs#L160), you can see the year has been updated in the test to get it passing again.

What I don't understand is why I don't get the binary package from cache.nixos.org.

Reply:
What Nixpkgs revision are you trying to build from?

Reply:
I don't know what a revision means or how to fetch that. I'm using the latest `nixos-22.11` channel.

Reply:
#213905 :)

Reply:
Nice one. Thank you.

Reply:
I've tested the config without having the `ExecPath` option. The app does not start in that case:
```
Feb 02 14:07:28 nixos systemd[56151]: jackett.service: Failed to locate executable /nix/store/mdwnvjcbq8hl8chw09f5r80havrq003>
░░ Subject: Process /nix/store/mdwnvjcbq8hl8chw09f5r80havrq0039-jackett-0.20.2782/bin/Jackett could not be executed
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ The process /nix/store/mdwnvjcbq8hl8chw09f5r80havrq0039-jackett-0.20.2782/bin/Jackett could not be executed and failed.
░░ 
░░ The error number returned by this process is ERRNO.
```

When I include both the package directory and Bash (`pkgs.jackett.outPath`, `pkgs.bash.outPath`), the error message changes:
```
Feb 02 14:18:15 nixos systemd[1]: Started Jackett.
░░ Subject: A start job for unit jackett.service has finished successfully
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ A start job for unit jackett.service has finished successfully.
░░ 
░░ The job identifier is 23144.
Feb 02 14:18:15 nixos systemd[1]: jackett.service: Main process exited, code=exited, status=203/EXEC
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░ 
░░ An ExecStart= process belonging to unit jackett.service has exited.
░░ 
░░ The process' exit code is 'exited' and its exit status is 203.
```

On the OS level this is the error:
```
execve("/nix/store/mdwnvjcbq8hl8chw09f5r80havrq0039-jackett-0.20.2782/bin/Jackett", ["/nix/store/mdwnvjcbq8hl8chw09f5r"..., "--NoUpdates", "--DataFolder", "/mnt/storage/jackett"], 0x55e98bafa200 /* 11 vars */) = -1 EACCES (Permission denied)
```

I have no idea why that is.

Reply:
I have discovered that Linux prevents loading shared libraries from filesystems which are mounted with the `noexec` option (Essentially that's what `NoExecPaths` does). That's the reason for the error above. It means all directories of `*.so` files must be added to the `ExecPaths` option.
To test this out, I added all directories listed by
`nix-store --query --requisites /nix/store/mdwnvjcbq8hl8chw09f5r80havrq0039-jackett-0.20.2782`
Now the Jackett service runs as expected.

So, I see two ways going forward:

1. Allow execution from the whole Nix store (current version)
2. Fetch all the runtime dependencies of Jackett and add those paths to `ExecPaths` (Unfortunately it's beyond my expertise).